sdk/communication/Azure.Communication.CallingServer/tests/MediaStreaming/CallAutomationStreamingParserTests.cs

```diff
@@ -0,0 +1,132 @@
+            jsonMetadata["format"]["sampleRate"] = 8;
+            jsonMetadata["format"]["channels"] = 2;
+            jsonMetadata["format"]["length"] = 100.1;
+
+            JObject jsonAudio = new JObject();
+            jsonAudio["data"] = "AQIDBAU=";
+            jsonAudio["timestamp"] = "2022-08-23T11:48:05Z";
+            jsonAudio["participantId"] = "participantId";
+            jsonAudio["isSilence"] = false;
+
+            JArray data = new JArray();
+            data.Add(jsonMetadata);
+            data.Add(jsonAudio);
+
+            BinaryData json = new BinaryData(data.ToString());
+            MediaStreamingPackageBase[] parsedPackages = MediaStreamingPackageParser.ParseMany(json);
+            Assert.NotNull(parsedPackages);
+            Assert.AreEqual(2, parsedPackages.Length);
+            Assert.IsTrue(parsedPackages[0] is MediaStreamingMetadata);
+            Assert.IsTrue(parsedPackages[1] is MediaStreamingAudio);
+            ValidateMetadata((MediaStreamingMetadata)parsedPackages[0]);
+            ValidateAudioData((MediaStreamingAudio)parsedPackages[1]);
+        }
+
+        [Test]
+        public void ParseAudioEventsWithBynaryArray()
+        {
+            JObject jsonAudio = new JObject();
+            jsonAudio["data"] = "AQIDBAU=";
+            jsonAudio["timestamp"] = "2022-08-23T11:48:05Z";
+            jsonAudio["participantId"] = "participantId";
+            jsonAudio["isSilence"] = false;
+
+            byte[] receivedBytes = System.Text.Encoding.UTF8.GetBytes(jsonAudio.ToString());
+            MediaStreamingAudio parsedPackage = (MediaStreamingAudio) MediaStreamingPackageParser.Parse(receivedBytes);
+
+            Assert.NotNull(parsedPackage);
+            ValidateAudioData(parsedPackage);
+        }
+
+        private static void ValidateMetadata(MediaStreamingMetadata streamingMetadata)
+        {
+            Assert.IsNotNull(streamingMetadata);
+            Assert.AreEqual("subscriptionId", streamingMetadata.MediaSubscriptionId);
+
+            ValidateFormat(streamingMetadata.Format);
+        }
+
+        private static void ValidateFormat(MediaStreamingFormat streamingFormat)
+        {
+            Assert.IsNotNull(streamingFormat);
+            Assert.AreEqual("encodingType", streamingFormat.Encoding);
+            Assert.AreEqual(8, streamingFormat.SampleRate);
+            Assert.AreEqual(2, streamingFormat.Channels);
+            Assert.AreEqual(100.1, streamingFormat.Length);
+        }
+
+        private static void ValidateAudioData(MediaStreamingAudio streamingAudio)
+        {
+            Assert.IsNotNull(streamingAudio);
+            Assert.AreEqual(5, streamingAudio.Data.ToArray().Length);
+            Assert.AreEqual(2022, streamingAudio.Timestamp.Year);
+            Assert.IsTrue(streamingAudio.Participant is CommunicationIdentifier);
+            Assert.AreEqual("participantId", streamingAudio.Participant.RawId);
+            Assert.IsFalse(streamingAudio.IsSilent);
```

Comment:
typo: IsSilence

Reply:
It can go two ways: AudioPackage is silent, or AudioPackage contains silence.
Let me know how you interpret it. :)

Reply:
yup, I saw that later so marked this resolved :)